src/hyperloglog.c

```diff
@@ -445,7 +446,99 @@ uint64_t MurmurHash64A (const void * key, int len, unsigned int seed) {
     h ^= h >> r;
     return h;
 }
+#else
+static inline void _wymum(uint64_t *A, uint64_t *B) {
```

Comment:
I appreciate the effort to enhance our codebase, but I would like us to exercise caution when importing code from external repositories. Integrating external code may introduce unforeseen challenges such as compatibility issues, licensing concerns, and divergent coding styles. Before incorporating such changes, let's thoroughly assess the impact and consider alternatives to ensure the long-term maintainability and stability of our project.

@yossigo @oranagra @itamarhaber @soloestoy @madolson @antirez - Any thoughts?

Reply:
Hi @oranagra, 

Thanks for your time to response.

My intention is try to optimize it from performance perspective, because according to the implementation of MurmurHash64A, it can't benefit from modern CPU arch (wider pipeline and more execution ports).  And I saw the MurmurHash64A has been the bottleneck in the hotspot profiling in related HLL benchmark tests.

I totally understand your concern of the backwards compatibility. You are right, I shouldn't let user choose at compile time. Do you think it is meaningful to solve the backwards compatibility in other approaches? Just as you suggested, use a separate rdb type to serve for the older version? When Redis load files from older version, keep MurmurHash64, use the new one if user started Redis from scratch etc.

Reply:
considering the ~20% performance gain, i think we should do it.
the only parts i'm not sure about is whether someone will have time to review this.
and how popular is HLL to justify investing time in this.

Reply:
Got it, I will continue this work and solve the backwards compatibility in this PR.  
You can review this when you are free.

Reply:
maybe you'd rather wait wait with investing more work on it till after we discuss this approach with the rest of the core-team.

Reply:
Sure, looking forward your approach. :)

Reply:
May be one more approach we can consider is:

Instead of copying code directly we can link the respective library, that way we don't have to worry about maintaining code. we'll have to just maintain library versions. This would also be quicker as many scanners such as black duck helps in ensuring vulnerabilities based on library versions.